vpr/src/route/rr_graph2.cpp

```diff
@@ -1082,33 +1082,102 @@ static void load_block_rr_indices(const DeviceGrid& grid,
-                        int x_tile = x + width_offset;
-                        for (int height_offset = 0; height_offset < type->height; ++height_offset) {
-                            int y_tile = y + height_offset;
-                            for (e_side side : SIDES) {
+                    bool assigned_to_rr_node = false;
+                    for (e_side side : wanted_sides) {
+                        for (int width_offset = 0; width_offset < type->width; ++width_offset) {
+                            int x_tile = x + width_offset;
+                            for (int height_offset = 0; height_offset < type->height; ++height_offset) {
+                                int y_tile = y + height_offset;
                                 if (type->pinloc[width_offset][height_offset][side][ipin]) {
                                     int iclass = type->pin_class[ipin];
                                     auto class_type = type->class_inf[iclass].type;
 
                                     if (class_type == DRIVER) {
                                         indices[OPIN][x_tile][y_tile][side].push_back(*index);
                                         indices[IPIN][x_tile][y_tile][side].push_back(OPEN);
+                                        assigned_to_rr_node = true;
                                     } else {
                                         VTR_ASSERT(class_type == RECEIVER);
-                                        indices[IPIN][x_tile][y_tile][side].push_back(*index);
                                         indices[OPIN][x_tile][y_tile][side].push_back(OPEN);
+                                        indices[IPIN][x_tile][y_tile][side].push_back(*index);
+                                        assigned_to_rr_node = true;
                                     }
-                                    ++(*index);
                                 } else {
                                     indices[IPIN][x_tile][y_tile][side].push_back(OPEN);
                                     indices[OPIN][x_tile][y_tile][side].push_back(OPEN);
                                 }
                             }
                         }
                     }
+                    /* A pin may locate on multiple sides of a tile.
+                     * Instead of allocating multiple rr_nodes for the pin,
+                     * we just create a rr_node and make it indexable on these sides
+                     * As such, we can avoid redundant rr_node to be allocated
+                     * and multiple nets to be mapped to the pin
+                     *
+                     * Considering that some pin could be just dangling, we do not need
+                     * to create a void rr_node for it.
+                     * As such, we only allocate a rr node when the pin is indeed located
+                     * on at least one side
+                     */
+                    if (true == assigned_to_rr_node) {
```

Comment:
The condition should just be `assigned_to_rr_node`.

Reply:
Hello, I am sorry I do not get the exact issue. Would you mind providing more insights? I will make a change.

Reply:
`if (assigned_to_rr_node)` is more clear than `if (true == assigned_to_rr_node)`.

There isn't a good reason to compare with a boolean literal.